### PR TITLE
Use consistently 1-D, 2-D, or 3-D in 'session.py' and 'conversion.py' (with hyphen and capitalized)

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR is similar to PR #2252, aiming to use consistently 1-D, 2-D, or 3-D (with hyphen and capitalized).

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by seisman — APPROVED** [review]
Looks good to me except my comment about `np.str_`.

File: pygmt/clib/session.py [hunk 1]
```diff
@@ -741,11 +741,11 @@ def put_vector(self, dataset, column, vector):
         first. Use ``family='GMT_IS_DATASET|GMT_VIA_VECTOR'``.
 
         Not all numpy dtypes are supported, only: int8, int16, int32, int64,
-        uint8, uint16, uint32, uint64, float32, float64, str\_ and datetime64.
+        uint8, uint16, uint32, uint64, float32, float64, str, and datetime64.
```
[the remark this answers is not in the document]
Comment (yvonnefroehlich) [reply pygmt/clib/session.py]: Thanks for sharing this link @seisman!

[the remark this answers is not in the document]
Comment (yvonnefroehlich) [reply pygmt/clib/session.py]: Thanks for sharing this link @seisman!

**Review by michaelgrund — APPROVED** [review]
Looks good!

**Pull request merged**